fixed-price-sale/program/src/processor/buy.rs

```diff
@@ -45,10 +42,97 @@ impl<'info> Buy<'info> {
         let system_program = &self.system_program;
 
         let metadata_mint = selling_resource.resource;
-        // do supply +1 to increase master edition supply
-        let edition = get_supply_off_master_edition(&master_edition.to_account_info())?
-            .checked_add(1)
-            .ok_or(ErrorCode::MathOverflow)?;
+
+        // First we find the edition marker number. If it's passed in by the user, we use that and check that
+        // it matches the edition marker PDA passed in.
+
+        // Otherwise we start at 0 and loop until we find the right edition marker.
+
+        let edition_marker_number = if let Some(edition_marker_number) = edition_marker_number {
+            let edition_marker_number_str = edition_marker_number.to_string();
+
+            let pda = Pubkey::find_program_address(
+                &[
+                    "metadata".as_bytes(),
+                    mpl_token_metadata::ID.as_ref(),
+                    metadata_mint.as_ref(),
+                    "edition".as_bytes(),
+                    edition_marker_number_str.as_bytes(),
+                ],
+                &mpl_token_metadata::ID,
+            )
+            .0;
+
+            if pda != *edition_marker_info.key {
+                return Err(ErrorCode::InvalidEditionMarkerAccount.into());
+            }
+
+            edition_marker_number
+        } else {
+            let mut edition_marker_number = 0u64;
+            let edition_marker_number_str = edition_marker_number.to_string();
+
+            loop {
+                let pda = Pubkey::find_program_address(
+                    &[
+                        "metadata".as_bytes(),
+                        mpl_token_metadata::ID.as_ref(),
+                        metadata_mint.as_ref(),
+                        "edition".as_bytes(),
+                        edition_marker_number_str.as_bytes(),
```

Comment:
This is using the same value in every iteration of the loop.

Reply:
Good catch! I'm ignoring the long-running tests because the CI machines can't handle minting that many editions which is why the CI passed with this issue. Maybe I can use account injection to write some tests to check for multiple PDAs instead.